src/components/description_list/description_list.tsx

```diff
@@ -87,28 +35,39 @@ export const EuiDescriptionList: FunctionComponent<
   type = 'row',
   ...rest
 }) => {
+  const euiTheme = useEuiTheme();
+  const styles = euiDescriptionListStyles(euiTheme);
+
+  const cssStyles = [styles.euiDescriptionList, styles[type], styles[align]];
+
   const classes = classNames(
     'euiDescriptionList',
-    type ? typesToClassNameMap[type] : undefined,
-    align ? alignmentsToClassNameMap[align] : undefined,
-    textStyle ? textStylesToClassNameMap[textStyle] : undefined,
+    { [`euiDescriptionList--${type}`]: type },
+    { [`euiDescriptionList--${align}`]: align },
+    { [`euiDescriptionList--${textStyle}`]: textStyle },
     {
       'euiDescriptionList--compressed': compressed,
     },
     className
   );
```

Comment:
It looks like there's really only a few instances of `--column` and 1 case of `--responsiveColumn` CSS overrides in Kibana, and most of them are only there to override our existing CSS selector specificity. We should be able to remove the selector for all cases but one, which we can replace with `[data-type="column"]` instead.
```suggestion
  const classes = classNames('euiDescriptionList', className);
```